src/hexkit/providers/s3/provider.py

```diff
@@ -19,6 +19,8 @@
 Utilities for testing are located in `./testutils.py`.
 """
 
+# ruff: noqa: PLR0913
```

Comment:
The default max-args setting for ruff.pylint is 5, but the number of ignores here suggest we should maybe increase it.